iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -37,8 +37,16 @@ inline vector<T, Capacity>::vector(const uint64_t count, const T& value)
 
 template <typename T, uint64_t Capacity>
 inline vector<T, Capacity>::vector(const uint64_t count)
-    : vector(count, T())
 {
+    if (count > Capacity)
+    {
+        std::cerr << "Attemting to initialize a vector with more elements than its capacity!" << std::endl;
```

Comment:
This is an easy recoverable error. We just reduce the amount of elements to capacity. But I adjusted the error message so that the user knows exactly what is going on.

Reply:
Ok understood. @MatthiasKillat What's your opinion on this?